### PR TITLE
dispatch events

### PR DESCRIPTION
**Pull request by johnrees**
maybe events should be props, idk

we should probably also keep area in the class like you had before, i'm just trying to keep the ui as simple as possible atm

### CODE REVIEW

**Comment by johnrees** [comment]
(it console.logs area, but will spam you when you resize a polygon)

File: index.html [hunk 1]
```diff
@@ -15,5 +15,19 @@
...
+
+      map.addEventListener("ready", (event) => {
+        console.log("map ready");
+      });
+      map.addEventListener("areaChange", ({ detail: area }) => {
+        console.debug({ area });
+      });
+      map.addEventListener("geojsonChange", ({ detail: geojson }) => {
+        console.debug({ geojson });
+      });
```
Comment (jessicamcinchak) [inline index.html:30]: oh, cool! :+1:

File: src/my-map.ts [hunk 2]
```diff
@@ -107,30 +113,46 @@ export class MyMap extends LitElement {
...
 
   // render the map
   render() {
     return html`<script src="https://cdn.polyfill.io/v2/polyfill.min.js"></script>
       <link rel="stylesheet" href="https://cdn.skypack.dev/ol@^6.6.1/ol.css" />
-      <div id="map" />
-      ${this.drawMode
-        ? html`<span id="area">${this.totalArea}</span>`
-        : null} `;
+      <div id="map" />`;
```
Comment (jessicamcinchak) [inline src/my-map.ts:142]: totally agree map should really only be returning the map -- while doing this I was thinking it'd be interesting to return the totalArea that we calculate _in_ map as it's own `@customElement("total-area")` span that's easy to embed separately later or just omit, but have no idea if that's possible or how'd it work yet!

Comment (johnrees) [reply src/my-map.ts:142]: oh yeah that's an interesting idea, would be fun to explore

**Pull request merged**